lib/lightning_web/controllers/webhooks_controller.ex

```diff
@@ -1,30 +1,41 @@
 defmodule LightningWeb.WebhooksController do
   use LightningWeb, :controller
 
+  alias Lightning.Extensions.RateLimiter
+  alias Lightning.Extensions.RateLimiting.Context
   alias Lightning.Workflows
   alias Lightning.WorkOrders
 
   require OpenTelemetry.Tracer
 
   @spec create(Plug.Conn.t(), %{path: binary()}) :: Plug.Conn.t()
   def create(conn, _params) do
-    case conn.assigns.trigger do
-      nil ->
-        conn |> put_status(:not_found) |> json(%{"error" => "Webhook not found"})
+    with %Workflows.Trigger{enabled: true, workflow: %{project_id: project_id}} =
+           trigger <- conn.assigns.trigger,
+         :ok <-
+           RateLimiter.limit_request(conn, %Context{project_id: project_id}, []) do
```

Comment:
I think we should 'hide' the Context struct behind the `limit_request` function, we can cast either a keyword list or map into this struct and validate/throw errors if needed.

It's not a deal breaker right now, but I can see these kinds of contracts finding their way into lots of parts of the app - and we'd do well to close the contract down a bit so we can lower the mental overhead of wrapping the desired functionality.